tide-core/src/middleware.rs

```diff
@@ -21,11 +21,12 @@ where
     }
 }
 
+// TODO (PR258): We don't want this exposed outside of tide repo
 /// The remainder of a middleware chain, including the endpoint.
 #[allow(missing_debug_implementations)]
 pub struct Next<'a, State> {
-    pub(crate) endpoint: &'a DynEndpoint<State>,
-    pub(crate) next_middleware: &'a [Arc<dyn Middleware<State>>],
+    pub endpoint: &'a DynEndpoint<State>,
```

Comment:
We can at least remove `pub` of fields by adding a constructor method.

Reply:
Just did that.